### ISSUE TITLE
subclasses of pathlib.PurePosixPath never call __init__ or __new__

### ISSUE THREAD

**Issue by fa65ff75-8434-45eb-83c3-7c7fb12e50e9** [python/cpython#85281]
BPO | [41109](https://bugs.python.org/issue41109)
--- | :---
Nosy | @pitrou, @websurfer5, @conchylicultor, @Louis-Vincent
PRs | <li>python/cpython#21920</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2020-06-25.02:51:23.231>
labels = ['type-bug', '3.8', '3.9', '3.10', '3.7', 'library']
title = 'subclasses of pathlib.PurePosixPath never call __init__ or __new__'
updated_at = <Date 2020-08-19.06:37:57.142>
user = 'https://github.com/conchylicultor'
```

bugs.python.org fields:
```python
activity = <Date 2020-08-19.06:37:57.142>
actor = 'Jeffrey.Kintscher'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2020-06-25.02:51:23.231>
creator = 'conchylicultor'
dependencies = []
files = []
hgrepos = []
issue_num = 41109
keywords = ['patch']
message_count = 12.0
messages = ['372297', '372298', '375030', '375031', '375041', '375047', '375162', '375332', '375333', '375334', '375364', '375638']
nosy_count = 4.0
nosy_names = ['pitrou', 'Jeffrey.Kintscher', 'conchylicultor', 'louis-vincent.boudre']
pr_nums = ['21920']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue41109'
versions = ['Python 3.6', 'Python 3.7', 'Python 3.8', 'Python 3.9', 'Python 3.10']
```

</p></details>

**Comment by barneygale** [comment]
I believe #100481 would address this.

**Comment by barneygale** [comment]
Fixed in Python 3.12 / #102789 / 11c302055a2aab5373c2829d1a8610a64e72ab24
